packages/experimental/surface/src/plugins/GithubMarkdownPlugin/GithubMarkdownPlugin.tsx

```diff
@@ -0,0 +1,65 @@
+//
+// Copyright 2023 DXOS.org
+//
+// import React from 'react';
+
+import React from 'react-router';
+
+import { Document } from '@braneframe/types';
+
+import { definePlugin, PluginDefinition, Surface } from '../../framework';
+import { isSpace } from '../SpacePlugin';
+import { MainAll } from './MainAll';
+import { MainOne } from './MainOne';
+
+// import { OctokitProvider } from './OctokitProvider';
+
+export const isDocument = (datum: any): datum is Document => {
+  return 'type' in datum && 'name' in datum.type && Document.type.name === datum.type.name;
+};
+
+export const GithubMarkdownPlugin: PluginDefinition = definePlugin({
+  meta: {
+    id: 'dxos:GithubMarkdownPlugin',
+  },
+  provides: {
+    // context: ({ children }) => {
+    //   return <OctokitProvider>{children}</OctokitProvider>;
+    // },
+    router: {
+      routes: () => [
+        {
+          path: '/document/:spaceSlug/:documentSlug',
+          element: (
+            <Surface
+              component='dxos:SplitViewPlugin/SplitView'
+              surfaces={{
+                sidebar: { component: 'dxos:ListViewPlugin/ListView' },
+                main: { component: 'dxos:GithubMarkdownPlugin/MainOne' },
+              }}
+            />
+          ),
+        },
+      ],
+    },
+    component: (arg: any) => {
+      if (Array.isArray(arg)) {
+        const [datum, surfaceType] = arg;
```

Comment:
This is the pattern I wanted to check in about — maybe I'm misunderstanding the graph concept, but I wanted to explicitly flag somehow that a plugin-component was getting rendered into a surface that considers itself `main`.